src/plugins/MoreSearchOperators.ts

```diff
@@ -0,0 +1,213 @@
+                    return true;
+                case "desc":
+                    match.setData("sortOrder", "desc");
+                    return true;
+                default:
+                    return false;
+            }
+        },
+        follows: ["FILTER_SORT_ORDER"],
+        queryKey: "sort_order",
+        mutable: true,
+        _dataKey: "sortOrder"
+    },
+    ANSWER_SORT_BY: {
+        componentType: "ANSWER",
+        regex: /\s*(relevance|timestamp)/i,
+        validator: function (match) {
+            let value = match.getMatch(1);
+            switch (value) {
+                case "relevance":
+                    match.setData("sortBy", "relevance");
+                    return true;
+                case "timestamp":
+                    match.setData("sortBy", "timestamp");
+                    return true;
+                default:
+                    return false;
+            }
+        },
+        follows: ["FILTER_SORT_BY"],
+        queryKey: "sort_by",
+        mutable: true,
+        _dataKey: "sortBy"
+    },
+    ANSWER_FILE_NAME: {
+        componentType: "ANSWER",
+        regex: /(?:\s*([^\s]+))/,
+        validator: function (match) {
+            match.setData("fileName", match.getMatch(1));
+            return true;
+        },
+        follows: ["FILTER_FILE_NAME"],
+        queryKey: "attachment_filename",
+        mutable: true,
+        _dataKey: "fileName"
+    },
+    ANSWER_FILE_TYPE: {
+        componentType: "ANSWER",
+        regex: /(?:\s*([^\s]+))/,
+        validator: function (match) {
+            match.setData("fileType", match.getMatch(1));
+            return true;
+        },
+        follows: ["FILTER_FILE_TYPE"],
+        queryKey: "attachment_extension",
+        mutable: true,
+        _dataKey: "fileType"
+    },
+    ANSWER_EMBED_TYPE: {
+        componentType: "ANSWER",
+        regex: /\s*([^\s]+)/i,
+        validator: function (match) {
+            match.setData("embedType", match.getMatch(1));
+            return true;
+        },
+        follows: ["FILTER_EMBED_TYPE"],
+        queryKey: "embed_type",
+        mutable: true,
+        _dataKey: "embedType"
+    },
+};
+
+export default definePlugin({
+    name: "MoreSearchOperators",
+    description: "Adds experimental search operators.",
+    authors: [Devs.Davri],
+
+    patches: [{
+        find: "Messages.SEARCH_SHORTCUT",
+        replacement: {
+            match: /(.)\((.),.\..{1,3}\.ANSWER_PINNED/,
+            replace: Object.keys(searchOperators).map(a => `$1($2,"${a}",$self.searchOperators.${a}),`).join("") + Object.keys(searchAnswers).map(a => `$1($2,"${a}",$self.searchAnswers.${a}),`).join("") + "$&"
+        }
+    }, {
+        find: "Messages.SEARCH_ANSWER_FROM",
+        replacement: {
+            match: /var .=.\[.\];switch\(.\)\{/,
+            replace: "$&" + Object.entries(searchAnswers).map(([k, v]) => `case "${k}":a.add(e.getData("${v['_dataKey']}"));break;`).join("")
```

Comment:
```suggestion
            match: /(?<=\i\.forEach\(\(function\((\i)\).{300,500})var (\i)=\i\[\i\];switch\(\i\)\{/,
            replace: "$&" + Object.entries(searchAnswers).map(([k, v]) => `case "${k}":$2.add($1.getData("${v['_dataKey']}"));break;`).join("")
```
You shouldn't hard code variable names in replaces. I used a lookbehind here to capture the search answer variable, and to not mess with the match replacement. Also using a limited wildcard match is kinda epic for things like this (eg: `.{200,500}`)